src/model_io/urdf/src/URDFModelExport.cpp

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (C) 2017 Fondazione Istituto Italiano di Tecnologia
+ * Authors: Silvio Traversaro
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include "URDFModelExport.h"
+
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Core/SpatialInertia.h>
+#include <iDynTree/Core/Transform.h>
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/Model/Traversal.h>
+#include <iDynTree/Model/FixedJoint.h>
+#include <iDynTree/Model/PrismaticJoint.h>
+#include <iDynTree/Model/RevoluteJoint.h>
+
+#include "URDFParsingUtils.h"
+
+
+#include <iomanip>
+#include <fstream>
+#include <string>
+
+#include <libxml/tree.h>
+#include <libxml/parser.h>
+
+namespace iDynTree
+{
+
+/**
+ * Add a <origin> URDF element to the specified parent element with the specified transform.
```

Comment:
if it's not an over-kill, it will be nice to document how this might look in an urdf file.

Reply:
You mean like adding an example of a origin tag, i.e.: 
~~~xml
<origin xyz="0 0 0.5" rpy="0 0 0"/>
~~~
?

Reply:
Something like that. It might be helpful for the developers to have a visual example of code in the documentation. Because sometimes, just looking at the arguments of the API can be ambiguous.

Reply:
Ack, I will do it. Consider anyhow that this function is not included in any public API, and so it is not directed to iDynTree users, but rather to people that will edit the exporter itself.

Reply:
Yes, I realized that. :)

Reply:
I had the same feeling when going through it, makes it even easier to review if you don't remember exactly at the moment the expected urdf format.